tests/Service/PromotionsService/AbstractPromotionServiceTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Tests\BBC\ProgrammesPagesService\Service\PromotionsService;
+
+use BBC\ProgrammesPagesService\Domain\Entity\Brand;
+use BBC\ProgrammesPagesService\Domain\Entity\Promotion;
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use BBC\ProgrammesPagesService\Service\PromotionsService;
+use Tests\BBC\ProgrammesPagesService\AbstractServiceTest;
+
+abstract class AbstractPromotionServiceTest extends AbstractServiceTest
+{
+    public function setUp()
+    {
+        $this->setUpCache();
+        $this->setUpRepo('PromotionRepository');
+        // override getDomainModel() in mapper with getDomainModelFromDbData()
+        $this->setUpMapper('PromotionMapper', 'getDomainModelFromDbData');
+    }
+
+    /**
+     * @param mixed[] $dbPromotion
+     */
+    protected function getDomainModelFromDbData(array $dbPromotion): Promotion
+    {
+        $pid = new Pid($dbPromotion['pid']);
+
+        $mockPromotion = $this->createMock(self::ENTITY_NS . 'Promotion');
+        $mockPromotion->method('getPid')->willReturn($pid);
+        return $mockPromotion;
+    }
+
+    /**
+     * @param array[] $dbPromotions
+     * @return Promotion[]
+     */
+    protected function getDomainModelsFromDbData(array $dbPromotions): array
+    {
+        return array_map([$this, 'getDomainModelFromDbData'], $dbPromotions);
+    }
+
+    protected function getMockProgramme(): Brand
+    {
+        $coreEntity = $this->createMock('BBC\ProgrammesPagesService\Domain\Entity\Brand');
```

Comment:
Is `$coreEntity` the best name for this? The method is called `getMockProgramme()`, you're mocking a `Brand` and calling it `$coreEntity`.  
What is it you're actually testing? Is it something that implements a `PromotableInterface`? If so, maybe rename some of these things to make it more obvious it's a `PromotableInterface` you're testing.